neqo-common/src/qlog.rs

```diff
@@ -89,10 +114,15 @@ impl Qlog {
     }
 
     /// If logging enabled, closure may generate an event to be logged.
+    #[cfg_attr(
+        not(feature = "qlog"),
+        expect(unused_variables, clippy::unused_self, reason = "Only used with qlog.")
+    )]
```

Comment:
This function is good.  You can't use `if cfg!(` because that means taking in the qlog crate dependency.